pkg/ossm/smcp_annotation.go

```diff
@@ -89,18 +89,31 @@ func TestSMCPAnnotations(t *testing.T) {
 			util.KubeApplyContents("bookinfo", testAnnotationProxyEnv)
 		}
 		util.CheckPodRunning("bookinfo", "app=env")
-		time.Sleep(time.Duration(10) * time.Second)
-
-		msg, _ := util.Shell(`kubectl get po -n bookinfo -l app=env -o yaml | grep "annotation-from-smcp"`)
-
-		if !strings.Contains(msg, `test1.annotation-from-smcp: test1`) {
-			t.Errorf("Failed to get annotations: %v", `test1.annotation-from-smcp: test1`)
+		podName, err := util.GetPodName("bookinfo", "app=env")
```

Comment:
You could apply retries like for annotations, but you can do it in a subsequent PR.